R/table.R

```diff
@@ -63,31 +63,49 @@ html_table.xml_node <- function(x, header = NA, trim = TRUE,
 
   ncols <- lapply(cells, html_attr, "colspan", default = "1")
   ncols <- lapply(ncols, as.integer)
+  nrows <- lapply(cells, html_attr, "rowspan", default = "1")
+  nrows <- lapply(nrows, as.integer)
+
   p <- unique(vapply(ncols, sum, integer(1)))
+  maxp <- max(p)
 
-  if (length(p) > 1) {
+  if (length(p) > 1 & (maxp * n != Reduce(sum, lapply(nrows, sum)))) {
```

Comment:
Good point; I'll refactor. Yep, equivalent to sum(unlist(nrows)) which is much cleaner. 
The logic here is the following:

If the true number of cells (which is maxp * n interpreting "true number of cells" to be the number of cells with repetition as the fill-in strategy) is different from the number of cells detected by looking at rowspan, then there is some filling to do. For instance, imagine this case:
<table>
<tr><th>A</th><th>B</th><th>C</th></tr>
<tr><td rowspan="2">Bob</td><td>Jane</td><td>Andy</td></tr>
<tr><td>Deb</td><td>Barb</td></tr>
</table>
which we would like to standardize as:
<table>
<tr><th>A</th><th>B</th><th>C</th></tr>
<tr><td>Bob</td><td>Jane</td><td>Andy</td></tr>
<tr><td>Bob</td><td>Deb</td><td>Barb</td></tr>
</table>
In this case maxp=3, n=3 and nrows=list(c(1,1,1), c(2,1,1), c(1,1)) so that Reduce(sum, lapply(nrows, sum)) = 9. With this change we know how to fill that rowspan so we don't even ask for fill=T. But, on the other hand, if the table is not well formed and that rowspan was left out, then we would have Reduce(sum, lapply(nrows, sum)) = 8 and so we would proceed as before, and only filling row 3, column 1 with NA if the user specifies fill=T. Then we would have the table below which is what currently happens with rowspan.
<table>
<tr><th>A</th><th>B</th><th>C</th></tr>
<tr><td>Bob</td><td>Jane</td><td>Andy</td></tr>
<tr><td>Deb</td><td>Barb</td></tr>
</table>